numpy/core/tests/test_function_base.py

```diff
@@ -29,6 +29,30 @@ def test_dtype(self):
         assert_equal(y.dtype, dtype('int32'))
 
 
+class TestGeomspace(TestCase):
+
+    def test_basic(self):
+        y = geomspace(1, 1e6)
+        assert_(len(y) == 50)
+        y = geomspace(1, 1e6, num=100)
+        assert_(y[-1] == 10 ** 6)
+        y = geomspace(1, 1e6, endpoint=False)
+        assert_(y[-1] < 10 ** 6)
+        y = geomspace(1, 1e6, num=7)
+        assert_array_equal(y, [1, 10, 100, 1e3, 1e4, 1e5, 1e6])
+
+    def test_dtype(self):
+        y = geomspace(1, 1e6, dtype='float32')
+        assert_equal(y.dtype, dtype('float32'))
+        y = geomspace(1, 1e6, dtype='float64')
+        assert_equal(y.dtype, dtype('float64'))
+        y = geomspace(1, 1e6, dtype='int32')
+        assert_equal(y.dtype, dtype('int32'))
+
```

Comment:
You may want to ask for the opinion of someone who might use such a feature. I can see myself using this function, but only with real inputs, so I would be fine with complex and negative inputs raising an error, returning nan or connecting the endpoints via some meaningful curve. The last option is probably optimal. The point of documenting and writing tests would be that if you change the internals, the public API remains the same and there are tests to make sure that happens.